docs/source/upgrade/index.rst

```diff
@@ -82,13 +83,109 @@ has one *nonconsuming* choice that accepts a *Coin* contract, archives
 this *Coin* contract and creates a *CoinWithAmount* contract with
 *amount* set to 1.
 
-.. literalinclude:: daml/Upgrade.daml
+.. literalinclude:: example/coinV2/daml/UpgradeFromCoinV1.daml
   :language: daml
   :start-after: -- UPGRADE_AGREEMENT_BEGIN
   :end-before: -- UPGRADE_AGREEMENT_END
 
-.. TODO Turn this into a full example project with multiple packages
-.. and explain builds and deployment.
 
-.. Building and deploying
-.. ======================
+Building and deploying version-1
+================================
+
+Let's see everything in action by following the steps to build and deploy version-1 of our coin. After this we'll see how to deploy and upgrade to version-2.
+
+First we'll need a sandbox ledger to which we can deploy.
+
+.. code-block:: none
+
+   $ daml sandbox --port 7001
+
+The version-1 project is named ``coinV1``. It contains just the original *Coin* definition template ``Coin``. Here is the ``daml.yaml`` setup
+
+.. literalinclude:: example/coinV1/daml.yaml
+  :language: yaml
+  :start-after: # BEGIN
+  :end-before: # END
+
+To make the example complete, we need a way to create some coins which have different parties for the issuer and owner. We take the standard approach of adding a :ref:`Propose-Accept workflow <intro propose accept>` to our example.
+
+.. literalinclude:: example/coinV1/daml/CoinV1.daml
+  :language: daml
+  :start-after: -- COIN_TRANSFER_BEGIN
+  :end-before: -- COIN_TRANSFER_END
+
+Now we can build and deploy version-1 of our Coin.
+
+.. code-block:: none
+
+   $ cd example/coinV1
+   $ daml build
+   $ daml deploy --port 7001
```

Comment:
Afaik `daml deploy` is `daml build + daml ledger upload-dar` so you can either skip the `build` step or use `ledger upload-dar`.

Reply:
Just off the top of my head, I think having the `build` step discrete is a good idea if this at some point goes in the tangent direction of ts generation, scripts and triggers.

Reply:
I prefer keeping the build and deploy steps explicit. I don`t think the example is clarified by taking advantage of the implicit `build` initiated by `deploy`.

Reply:
Maybe just switch to `daml ledger upload-dar` then? IIRC `daml deploy` also does party allocation in a somewhat sketchy way so I think it actually does more harm than good in most cases.